Show default size-cutoff value in --help output

RyanGlScott opened the pull request:
It wasn't clear to me from the current `--help` output that the default `--size-cutoff` value is 1000 bytes. This patch tweaks the `--help` output to make this explicit.

feuerbach commented:
Good point. optparse-applicative actually has a feature to show the default value: https://hackage.haskell.org/package/optparse-applicative-0.15.1.0/docs/Options-Applicative.html#v:showDefaultWith

RyanGlScott commented:
Should I interpret https://github.com/feuerbach/tasty-golden/pull/33#issuecomment-596004067 to mean that you'd prefer displaying the default value like this instead?

```diff
diff --git a/Test/Tasty/Golden/Internal.hs b/Test/Tasty/Golden/Internal.hs
index 70f0030..4ca3cdb 100644
--- a/Test/Tasty/Golden/Internal.hs
+++ b/Test/Tasty/Golden/Internal.hs
@@ -8,7 +8,7 @@ import Data.Typeable (Typeable)
 import Data.Proxy
 import Data.Int
 import System.IO.Error (isDoesNotExistError)
-import Options.Applicative (metavar)
+import Options.Applicative (metavar, showDefaultWith, value)
 import Test.Tasty.Providers
 import Test.Tasty.Options
 
@@ -49,14 +49,17 @@ instance IsOption NoCreateFile where
 -- for readability.
 --
 -- The default value is 1000 (i.e. 1Kb).
-newtype SizeCutoff = SizeCutoff Int64
+newtype SizeCutoff = SizeCutoff { getSizeCutoff :: Int64 }
   deriving (Eq, Ord, Typeable, Num, Real, Enum, Integral)
 instance IsOption SizeCutoff where
   defaultValue = 1000
   parseValue = fmap SizeCutoff . safeRead . filter (/= '_')
   optionName = return "size-cutoff"
   optionHelp = return "hide golden test output if it's larger than n bytes"
-  optionCLParser = mkOptionCLParser $ metavar "n"
+  optionCLParser = mkOptionCLParser $
+    metavar "n" <>
+    value defaultValue <>
+    showDefaultWith (show . getSizeCutoff)
 
 instance IsTest Golden where
   run opts golden _ = runGolden golden opts
```

If so, I would be fine with that. I only went with the current approach since that's how `tasty` does it (see the [`optionHelp` for `UseColor`](https://github.com/feuerbach/tasty/blob/5e4000f370f6d3d483439bf003ab925d43630b4b/core/Test/Tasty/Ingredients/ConsoleReporter.hs#L461), for instance).

feuerbach commented:
> Should I interpret #33 (comment) to mean that you'd prefer displaying the default value like this instead?

Yes please!

> If so, I would be fine with that. I only went with the current approach since that's how tasty does it (see the optionHelp for UseColor, for instance).

Yes, we should use that combinator in tasty itself as well. The reason (excuse) why it's not in use right now is that it's a relatively new feature and didn't exist at the time (I think).

RyanGlScott commented:
I've pushed an update which uses `showDefaultWith`. According the [changelog for `optparse-applicative`](https://hackage.haskell.org/package/optparse-applicative-0.15.1.0/changelog), `showDefaultWith` was introduced in version 0.3.1 (released in 2012), so I have bumped the lower version bounds on `optparse-applicative` accordingly.

feuerbach commented:
Perfect, just one last thing: could you import `<>` via Data.Monoid, thus avoiding the CPP and the conditional dependency on semigroups?

RyanGlScott commented:
I switched the import to use `Data.Monoid` and removed the conditional `semigroups` dependency. That being said, I wasn't able to remove the CPP. Importing `(<>)` on recent GHCs will give a redundant import warning, since `(<>)` is also exported by the `Prelude`. To avoid this warning, I guarded the `Data.Monoid ((<>))` import with `#if !MIN_VERSION_base(4,11,0) ... #endif`.

feuerbach commented:
Ah, fair enough. Thanks!
